### PR TITLE
Fix summary link naming

### PR DESCRIPTION
**Pull request by adityasaky**
Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

**Fixes issue #**: #254 

**Description of the changes being introduced by the pull request**:

I've implemented the same fix used in the golang repository - see in-toto/in-toto-golang#19.

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
What is a "summary link?" First time I ever heard about it.

**Comment by adityasaky** [comment]
Summary links are used during sublayout verification  - a link summarizes the steps in the sublayout and takes its place. Right now, the name of a summary link is set as the name of the first link of the sublayout. When I was working on in-toto/in-toto-golang, @lukpueh and I realized this could be a potential problem, and this is a simple fix to use the corresponding step name instead. See: #254

File: in_toto/verifylib.py [hunk 1]
```diff
@@ -1455,6 +1455,9 @@ def get_summary_link(layout, reduced_chain_link_dict):
               <link name> : <Metablock containing a Link object>,
               ...
             }
+    step_name:
+            The step that the layout corresponds to, only used in sublayout
+            verification.
```
Comment (lukpueh) [inline in_toto/verifylib.py]: Minor suggestions: I think the description and the name give more context about the *usual* way of calling `get_summary_link` than necessary. I'd just call the parameter `name` or `link_name` and  say that it is the name assigned to the link.

File: in_toto/verifylib.py [hunk 2]
```diff
@@ -1568,6 +1571,9 @@ def in_toto_verify(layout, layout_key_dict, link_dir_path=".",
               - artifact rules in step and inspection definitions in the layout
               - the run fields in the inspection definitions
               - the expected command in the step definitions
+    step_name: (optional)
+            The step that the layout corresponds to, used only for sublayout
+            verification to generate summary links.
```
Comment (lukpueh) [inline in_toto/verifylib.py]: Maybe it is worth pointing out that the parameter is used in recursive calls of `in_toto_verify` and to refer the reader to `verify_sublayouts` to see how it is used?

**Review by lukpueh — CHANGES_REQUESTED** [review]
Thanks for the PR, @adityasaky! Please consider addressing my inline suggestions. Also, would you mind adding a test, where a MATCH rule of a super layout refers to the step, represented by the (now correctly named) summary link of a sublayout? (see #254)

**Review by SantiagoTorres — APPROVED** [review]
LGTM

**Comment by lukpueh** [comment]
Thanks for addressing my concerns, @adityasaky. Can you please add the requested test?

**Comment by adityasaky** [comment]
Yep, I'm working on that @lukpueh.

**Comment by adityasaky** [comment]
I think this suffices, @lukpueh. Please review and let me know!

**Review by lukpueh — APPROVED** [review]

**Comment by lukpueh** [comment]
Thanks for the awesome work, @adityasaky! In addition to reviewing your code I ran the test in debug mode and all looks good to me.

**Pull request merged**